### PR TITLE
[#175] Kompatibilität mit älteren Browsern

### PR DESCRIPTION
**Pull request by bitboxer**
closes https://github.com/railslove/recover-backlog/issues/175
closes https://github.com/railslove/recover-backlog/issues/164

### CODE REVIEW

File: .browserslistrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+> 0.5%
+last 2 versions
+Chrome >= 60
+ios_saf >= 10.0
+Firefox ESR
+not dead
```
Comment (bitboxer) [inline .browserslistrc:6]: If someone wants more older browsers supported, please say so 😉

File: pages/_app.tsx [hunk 2]
```diff
@@ -10,11 +11,22 @@ import { AnimateSharedLayout } from 'framer-motion'
...
 
 const App: React.FC<AppProps> = ({ Component, pageProps }) => {
   useA11yFocusRing()
+
+  if (
+    typeof navigator != 'undefined' &&
+    !supportedBrowsers.test(navigator.userAgent)
+  ) {
+    alert(
+      'Ihr Browser wird nicht unterstützt. Die Seite wird eventuell nicht funktionieren.'
```
Comment (bitboxer) [inline pages/_app.tsx:26]: We need to decide if this is the correct way and wording for us to show this warning.

File: .browserslistrc (hunk 1 above)
Comment (stringparser) [reply .browserslistrc:6]: Nice. Looks good 💪

File: README.md [hunk 3]
```diff
@@ -118,6 +118,11 @@ _[recover care](https://care.rcvr.app/)_ is a version of recover for institution
 BUILD_VARIANT=care npm run dev
 ```
 
+### Updating supported browsers
+
+After you updated the `.browserslistrc` you need to run `npm run supportedBrowsers` to update the
+regular expression the app is using to figure out if a browser is supported.
```
Comment (stringparser) [inline README.md:124]: Do you think we could have this generated bit `next.config.js` like

```js
webpack(config, { isServer }) {

    if (isServer) {
      updateSupportedBrowsers()
    }

    return config
},
```
or maybe just add `npm run supportedBrowsers` to the `build` script in `package.json`.

File: babel.config.js [hunk 4]
```diff
@@ -8,6 +16,9 @@ module.exports = function (api) {
       {
         autoLabel: 'dev-only',
         labelFormat: '[local]',
+        'preset-env': {
+          targets: browsersList,
```
Comment (stringparser) [inline babel.config.js:20]: Would also be nice to have [`useBuiltins: 'usage'`](https://babeljs.io/docs/en/babel-preset-env#usebuiltins) here I think

File: pages/_app.tsx (hunk 2 above)
Comment (stringparser) [reply pages/_app.tsx:26]: Looks good from the just inform the user but yes I'm not sure if we should so do something else. I'll think and research a bit about it.

File: babel.config.js (hunk 4 above)
Comment (bitboxer) [reply babel.config.js:20]: Ah, yes. Will do.

File: README.md (hunk 3 above)
Comment (bitboxer) [reply README.md:124]: Yes, you are right. Let me check where the best place would be.

File: pages/_app.tsx (hunk 2 above)
Comment (bitboxer) [reply pages/_app.tsx:26]: Kira wanted to investigate the exact wording here. Waiting for her input on this.

File: babel.config.js (hunk 4 above)
Comment (bitboxer) [reply babel.config.js:20]: And this was the reason why I needed the polyfil file. Thanks. Removed it and now will use babel 🥳

File: pages/_app.tsx (hunk 2 above)
Comment (stringparser) [reply pages/_app.tsx:26]: Thats nice :) 

From the implementation side I have no more comments. One way or another lead to this for what I've checked.

File: next.config.js [hunk 5]
```diff
@@ -1,8 +1,12 @@
+const execSync = require('child_process').execSync
+
 module.exports = {
   typescript: {
     ignoreDevErrors: true,
   },
   webpack: (config) => {
+    execSync('npm run update-supported-browsers')
```
Comment (stringparser) [inline next.config.js:8]: Nice so it works without checking for `isSever`. Just run it locally and was fine. 

Should we also add `lib/supportedBrowsers.js` to `.gitignore`?

Comment (stringparser) [reply next.config.js:8]: Or maybe put it under `lib/generated` and `.gitignore` that. I've also have some types on the `i18n` branch generated this way that I'm tinkering to figure a better way to generate without having to add them to the repo. 

Minor thing in my opinion though 😆 Just the usual thinking too much 😄

Comment (bitboxer) [reply next.config.js:8]: Ah, yes. With this we now can. Will update

**Review by stringparser — APPROVED** [review]

**Pull request merged**